### PR TITLE
fix: restore line expansion for multi-repo diffs

### PR DESCRIPTION
**Pull request by zeval**
Multi-repo tasks fetched expansion context from the non-Git task root and branch worktrees could lose their exact base ref. Expansion now resolves content in the owning repository and preserves each cumulative diff's base SHA through both review surfaces.

## Important Changes

- Route old-content reads through the repository-specific workspace tracker.
- Carry exact per-file base refs into Review and Changes diff rendering.
- Hide expansion controls when full-content reparsing is rejected.

## Validation

- `make fmt`
- `make typecheck`
- `make test`
- `make lint`
- Playwright: expand-arrow click, cumulative Review diff, multi-repo grouping, and mobile Review smoke
- No public docs change: internal diff-loading fix with no new user-facing workflow or configuration

## Possible Improvements

Low risk; PR-source expansion remains intentionally disabled and multi-PR Review still loads only the primary PR.

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.
- [ ] I checked whether this affects public docs in `docs/public/**` and updated them or noted why no docs change is needed.

### Desktop

Expanded context between two diff hunks.

Desktop line expansion

### Mobile

Expanded context in the native mobile diff sheet.

Mobile line expansion

### CODE REVIEW

File: apps/web/components/review/review-dialog.build-files.test.ts [hunk 1]
```diff
@@ -63,8 +63,16 @@ describe("buildAllFiles (review dialog)", () => {
...
+    expect(result).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({ path: "src/a.ts", base_ref: "abc" }),
+        expect.objectContaining({ path: "src/b.ts", base_ref: "abc" }),
+      ]),
+    );
   });
+});
 
+describe("buildAllFiles cumulative diff composition", () => {
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/components/review/review-dialog.build-files.test.ts]: Fixed in f0bdfd87f by renaming the block to the broader buildAllFiles label.

File: apps/backend/internal/agentctl/server/api/workspace.go [hunk 2]
```diff
@@ -154,13 +154,13 @@ func (s *Server) handleFileContentAtRef(c *gin.Context) {
 		return
 	}
 
-	scopedPath, err := s.procMgr.JoinRepoPath(c.Query("repo"), path)
+	tracker, err := s.procMgr.GetWorkspaceTrackerFor(c.Query("repo"))
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/backend/internal/agentctl/server/api/workspace.go]: Fixed in b97413e70. Multi-repo ref reads now require an explicit repository scope, with a regression test for repo-less requests.

File: apps/web/components/review/review-diff-list.tsx [hunk 3]
```diff
@@ -389,8 +389,9 @@ function useScrollIntoViewOnSelect(
  * (controls-less) render:
  *
  *  - uncommitted rows: diff is working-tree-vs-HEAD, so baseRef="HEAD".
- *  - committed rows: diff is base-branch-vs-HEAD, so baseRef is the repo's
- *    base branch. HEAD already contains the commits, so expanding against it
+ *  - committed rows: diff is base-vs-HEAD, so baseRef is the exact old-side
+ *    commit carried by the file. Legacy payloads fall back to the repo's base
+ *    branch. HEAD already contains the commits, so expanding against it
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/components/review/review-diff-list.tsx]: Fixed in b97413e70. The comment now says the exact file base ref is preferred and the repo base branch is only the fallback.
